### PR TITLE
Feature/8 scheduled backups

### LINKED ISSUE #8: Add scheduled backups

### PR DESCRIPTION
**Pull request by meiserloh**
Resolves #8

### CODE REVIEW

File: pkg/backupschedule/createManager.go [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+	_, err = backupScheduleClient.UpdateStatusCreated(ctx, backupSchedule)
+	if err != nil {
+		return fmt.Errorf("failed to set status [%s] in backup schedule resource [%s]: %w", v1.BackupScheduleStatusCreated, backupScheduleName, err)
+	}
+
+	return nil
+}
+
+func (cm *defaultCreateManager) createCronJob(ctx context.Context, schedule *v1.BackupSchedule) error {
```
Comment (cesmarvin) [inline pkg/backupschedule/createManager.go:59]: 🚫 **[checkstyle]** <funlen> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Function 'createCronJob' is too long (65 > 60)

File: pkg/additionalimages/updater.go [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+func (bsp *updater) patchCronJob(ctx context.Context, schedule *v1.BackupSchedule) error {
+	cronJobClient := bsp.clientSet.BatchV1().CronJobs(bsp.namespace)
+	cronJob, err := cronJobClient.Get(ctx, schedule.CronJobName(), metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to get cron job %s: %w", schedule.CronJobName(), err)
+	}
+
+	cronJob.Spec.JobTemplate.Spec.Template.Spec.Containers[0].Image = bsp.kubectlImage
+	cronJob, err = cronJobClient.Update(ctx, cronJob, metav1.UpdateOptions{})
```
Comment (cesmarvin) [inline pkg/additionalimages/updater.go:64]: 🚫 **[checkstyle]** <staticcheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>SA4006: this value of `cronJob` is never used

File: pkg/additionalimages/updater.go [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+		message := fmt.Sprintf("Cron job %s for backup schedule %s does not exist. Skipping kubectl image update.", schedule.CronJobName(), schedule.Name)
+		logger.Error(err, message)
+		bsu.recorder.Event(schedule, corev1.EventTypeWarning, imageUpdateEventReason, message)
+		return nil
+	} else if err != nil {
+		return fmt.Errorf("failed to get cron job %s: %w", schedule.CronJobName(), err)
+	}
+
+	cronJob.Spec.JobTemplate.Spec.Template.Spec.Containers[0].Image = bsu.kubectlImage
+	cronJob, err = cronJobClient.Update(ctx, cronJob, metav1.UpdateOptions{})
```
Comment (cesmarvin) [inline pkg/additionalimages/updater.go:100]: 🚫 **[checkstyle]** <staticcheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>SA4006: this value of `cronJob` is never used

File: pkg/additionalimages/updater.go [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
```
Comment (cesmarvin) [inline pkg/additionalimages/updater.go]: 🚫 **[checkstyle]** <staticcheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>SA4006: this value of `cronJob` is never used

**Review by jelemux — APPROVED** [review]
LGTM if CI is successful!

**Review by jelemux — APPROVED** [review]

**Comment by jelemux** [comment]
Gonna ignore unstable CI because we can't push test coverage any higher.

**Pull request merged**